cmd/osm-controller/osm-controller.go

```diff
@@ -229,6 +250,43 @@ func main() {
 	httpServer := httpserver.NewHTTPServer(funcProbes, httpProbes, metricsStore, constants.MetricsServerPort, debugServer)
 	httpServer.Start()
 
+	//Setting up k8s controller manager to reconcile OSM resources
+	log.Info().Msg("Setting up controller to reconcile OSM resources")
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
+		LeaderElection:     enableLeaderElection,
+		Port:               9443,
+		Namespace:          osmNamespace,
+	})
+	if err != nil {
+		log.Error().Err(err).Msg("Error starting up controller manager")
+		os.Exit(1)
+	}
+
+	log.Info().Msg("Successfully setup controller for reconcile")
+	log.Info().Msg("Setting up mutatingWebhookConfiguration reconciler")
+
+	// Adding a reconciler for OSM's mutatingwehbookconfiguration
+	if err = (&reconciler.MutatingWebhookConfigrationReconciler{
+		Client:       mgr.GetClient(),
+		Scheme:       mgr.GetScheme(),
+		OsmWebhook:   fmt.Sprintf("osm-webhook-%s", meshName),
+		OsmNamespace: osmNamespace,
+		CertManager:  certManager,
+	}).SetupWithManager(mgr); err != nil {
+		log.Error().Err(err).Msg("Error creating reconcile controller for MutatingWebhookConfiguration")
+		os.Exit(1)
+	}
+	// +kubebuilder:scaffold:builder
+
+	log.Info().Msg("starting manager")
+	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		log.Error().Err(err).Msg("problem running manager")
+		os.Exit(1)
+	}
+	log.Info().Msg("Successfully running controller manager")
```

Comment:
Would be better to move this initialization to the a separate function.

Reply:
done